sql: TestIdleCancelSession failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/3e9a4e9bc5849729c17d686ae83f647024338fc0

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stressrace instead of stress and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
make stress TESTS=TestIdleCancelSession PKG=github.com/cockroachdb/cockroach/pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-stderr=false -maxtime 20m -timeout 10m'
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=988013&tab=buildLog

```
I181026 12:07:40.682216 89311 server/node.go:722  [n?] node connected via gossip and verified as part of cluster "2d93ae8c-8957-4c02-92d7-21d7bab04453"
I181026 12:07:40.687636 89951 gossip/server.go:232  [n1] received initial cluster-verification connection from {tcp 127.0.0.1:44899}
I181026 12:07:40.735094 89311 server/node.go:425  [n?] new node allocated ID 2
I181026 12:07:40.735878 89311 gossip/gossip.go:395  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:44899" > attrs:<> locality:<> ServerVersion:<major_val:2 minor_val:1 patch:0 unstable:0 > build_tag:"v2.1.0-rc.2-4-g3e9a4e9" started_at:1540555660735405357 
I181026 12:07:40.737233 89311 storage/stores.go:242  [n2] read 0 node addresses from persistent storage
I181026 12:07:40.737746 89311 storage/stores.go:261  [n2] wrote 1 node addresses to persistent storage
I181026 12:07:40.746260 89570 storage/stores.go:261  [n1] wrote 1 node addresses to persistent storage
I181026 12:07:40.799260 89311 server/node.go:672  [n2] bootstrapped store [n2,s2]
I181026 12:07:40.802209 89311 server/node.go:546  [n2] node=2: started with [<no-attributes>=<in-mem>] engine(s) and attributes []
I181026 12:07:40.808472 89311 server/status/recorder.go:609  [n2] available memory from cgroups (8.0 EiB) exceeds system memory 16 GiB, using system memory
I181026 12:07:40.808733 89311 server/server.go:1816  [n2] Could not start heap profiler worker due to: directory to store profiles could not be determined
I181026 12:07:40.819422 89311 server/server.go:1525  [n2] starting https server at 127.0.0.1:45779 (use: 127.0.0.1:45779)
I181026 12:07:40.819712 89311 server/server.go:1527  [n2] starting grpc/postgres server at 127.0.0.1:44899
I181026 12:07:40.819818 89311 server/server.go:1528  [n2] advertising CockroachDB node at 127.0.0.1:44899
I181026 12:07:40.841647 89311 server/server.go:1581  [n2] done ensuring all necessary migrations have run
I181026 12:07:40.841987 89311 server/server.go:1584  [n2] serving sql connections
I181026 12:07:40.995643 90055 sql/event_log.go:126  [n2] Event: "node_join", target: 2, info: {Descriptor:{NodeID:2 Address:{NetworkField:tcp AddressField:127.0.0.1:44899} Attrs: Locality: ServerVersion:2.1 BuildTag:v2.1.0-rc.2-4-g3e9a4e9 StartedAt:1540555660735405357 LocalityAddress:[]} ClusterID:2d93ae8c-8957-4c02-92d7-21d7bab04453 StartedAt:1540555660735405357 LastUp:1540555660735405357}
I181026 12:07:41.001258 90053 server/server_update.go:67  [n2] no need to upgrade, cluster already at the newest version
I181026 12:07:41.141154 89613 gossip/gossip.go:1513  [n1] node has connected to cluster via gossip
I181026 12:07:41.146170 89613 storage/stores.go:261  [n1] wrote 1 node addresses to persistent storage
I181026 12:07:41.323178 90120 util/stop/stopper.go:537  quiescing; tasks left:
1      node.Node: batch
1      [async] transport racer
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I181026 12:07:41.323827 90121 util/stop/stopper.go:537  quiescing; tasks left:
1      node.Node: writing summary
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I181026 12:07:41.338716 90120 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] transport racer
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
W181026 12:07:41.342448 90051 server/node.go:845  [n2,summaries] error recording initial status summaries: failed to send RPC: sending to all 1 replicas failed; last error: <nil> rpc error: code = Unavailable desc = transport is closing
I181026 12:07:41.342640 90121 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I181026 12:07:41.348754 90120 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] transport racer
1      [async] closedts-subscription
I181026 12:07:41.349278 90120 util/stop/stopper.go:537  quiescing; tasks left:
1      [async] transport racer
I181026 12:07:41.353326 89504 kv/transport_race.go:91  transport race promotion: ran 26 iterations on up to 302 requests
--- FAIL: TestIdleCancelSession (2.47s)
	run_control_test.go:321: expected 2 sessions but found 1

```

This fails almost immediately under stress:

```
~  make stressrace PKG=./sql TESTS=TestIdleCancelSession
...
--- FAIL: TestIdleCancelSession (3.06s)
	run_control_test.go:321: expected 2 sessions but found 1
FAIL


ERROR: exit status 1

1 runs completed, 1 failures, over 3s
```

--- a/pkg/sql/run_control_test.go
+++ b/pkg/sql/run_control_test.go
@@ -302,7 +302,7 @@ func testCancelSession(t *testing.T, hasActiveSession bool) {
 	}
 
 	// Wait for node 2 to know about both sessions.
-	if err := retry.ForDuration(250*time.Millisecond, func() error {
+	if err := retry.ForDuration(10*time.Second, func() error {
 		rows, err := conn2.QueryContext(ctx, "SHOW CLUSTER SESSIONS")
 		if err != nil {
 			return err

